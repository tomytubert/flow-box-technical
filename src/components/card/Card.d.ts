import PropTypes from 'prop-types';
interface Props {
    image: string;
    altImage: string;
    avatarImage: string;
    avatarName: string;
    likes: number;
}
export declare const CardPost: {
    ({ image, avatarImage, avatarName, altImage, likes, }: Props): import("react/jsx-runtime").JSX.Element;
    propTypes: {
        image: PropTypes.Validator<string>;
        avatarImage: PropTypes.Validator<string>;
        avatarName: PropTypes.Validator<string>;
        altImage: PropTypes.Validator<string>;
        likes: PropTypes.Validator<number>;
    };
};
export {};
