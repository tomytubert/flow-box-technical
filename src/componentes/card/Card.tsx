import {
	Avatar,
	Box,
	Card,
	CardContent,
	CardHeader,
	CardMedia,
	Typography,
} from '@mui/material'
import FavoriteIcon from '@mui/icons-material/Favorite';

interface Props {
	image: string
	altImage: string
	avatarImage: string
	avatarName: string
	likes: number
}
export const CardPost = ({
	image,
	avatarImage,
	avatarName,
	altImage,
    likes
}: Props) => {
	return (
		<Card sx={{ width: 345 }}>
			<CardHeader
				avatar={<Avatar aria-label='recipe' src={avatarImage} />}
				title={`@${avatarName}`}
			/>
			<CardMedia component='img' height='194' image={image} alt={altImage} />
			<CardContent sx={{ height: '120px' }}>
                <Box display="flex" alignItems="center" gap={2} mb={1}>
					<FavoriteIcon />
                    <Typography variant='body2' color='text.secondary'>
                        {likes}
                    </Typography>
                </Box>
				<Typography variant='body2' color='text.secondary'>
					{altImage}
				</Typography>
			</CardContent>
		</Card>
	)
}
