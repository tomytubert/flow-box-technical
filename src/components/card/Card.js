import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { Avatar, Box, Card, CardContent, CardHeader, CardMedia, Typography, } from '@mui/material';
import FavoriteIcon from '@mui/icons-material/Favorite';
import PropTypes from 'prop-types';
export var CardPost = function (_a) {
    var image = _a.image, avatarImage = _a.avatarImage, avatarName = _a.avatarName, altImage = _a.altImage, likes = _a.likes;
    return (_jsxs(Card, { sx: { width: 345 }, children: [_jsx(CardHeader, { avatar: _jsx(Avatar, { "aria-label": 'recipe', src: avatarImage }), title: "@".concat(avatarName) }), _jsx(CardMedia, { component: 'img', height: '194', image: image, alt: altImage, loading: 'lazy' }), _jsxs(CardContent, { sx: { height: '120px' }, children: [_jsxs(Box, { display: 'flex', alignItems: 'center', gap: 2, mb: 1, children: [_jsx(FavoriteIcon, { color: 'primary' }), _jsx(Typography, { variant: 'body2', color: 'text.secondary', children: likes })] }), _jsx(Typography, { variant: 'body2', color: 'text.secondary', className: 'capitalize-first-letter', children: altImage })] })] }));
};
CardPost.propTypes = {
    image: PropTypes.string.isRequired,
    avatarImage: PropTypes.string.isRequired,
    avatarName: PropTypes.string.isRequired,
    altImage: PropTypes.string.isRequired,
    likes: PropTypes.number.isRequired,
};
