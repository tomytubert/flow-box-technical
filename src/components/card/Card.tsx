import {
	Avatar,
	Box,
	Card,
	CardContent,
	CardHeader,
	CardMedia,
	Typography,
} from '@mui/material'
import FavoriteIcon from '@mui/icons-material/Favorite'

interface CardPostProps {
	image: string
	altImage: string
	avatarImage: string
	avatarName: string
	likes: number
	isRowDisplay?: boolean
}
export const CardPost = ({
	image,
	avatarImage,
	avatarName,
	altImage,
	likes,
	isRowDisplay,
}: CardPostProps) => {
	return (
		<Card
			sx={{
				width: isRowDisplay ? '100%' : 345,
				display: isRowDisplay ? 'flex' : 'initial',
			}}
		>
			{!isRowDisplay && (
				<CardHeader
					avatar={<Avatar aria-label='recipe' src={avatarImage} />}
					title={`@${avatarName}`}
				/>
			)}
			<CardMedia
				component='img'
				height='194'
				image={image}
				alt={altImage}
				loading='lazy'
				sx={
					isRowDisplay
						? { flex: { md: 2 }, width: '200px', minWidth: '200px' }
						: {}
				}
			/>
			<CardContent
				sx={
					isRowDisplay
						? { height: { md: '120px' }, flex: { md: 2 } }
						: { height: '120px' }
				}
			>
				<Box display='flex' alignItems='center' gap={2} mb={1}>
					<FavoriteIcon color='primary' />
					<Typography variant='body2' color='text.secondary'>
						{likes}
					</Typography>
				</Box>
				<Typography
					variant='body2'
					color='text.secondary'
					className='capitalize-first-letter'
				>
					{altImage}
				</Typography>
			</CardContent>
		</Card>
	)
}
