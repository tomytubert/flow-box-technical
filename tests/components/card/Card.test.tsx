import React from 'react'
import { describe, expect, test } from '@jest/globals'
import { render, screen } from '@testing-library/react'
import { CardPost } from '../../../src/components/card/Card'

describe('Testing <Card />', () => {
	const post = {
		image: 'https://example.com/image.jpg',
		avatarImage: 'https://example.com/avatar.jpg',
		avatarName: 'example',
		altImage: 'example',
		likes: 100,
	}
	test('should match the snapshot', () => {
		const { container } = render(<CardPost {...post} />)
		expect(container).toMatchSnapshot()
	})
	test('should render the image', () => {
		render(<CardPost {...post} />)

		const { src, alt } = screen.getByAltText('example') as HTMLImageElement
		expect(src).toBe(post.image)
		expect(alt).toBe(post.altImage)
	})
	test('should render the avatar image', () => {
		render(<CardPost {...post} />)
		const avatarDiv = screen.getByLabelText('recipe')
		const { src } = avatarDiv.querySelector('img') as HTMLImageElement
		expect(src).toBe(post.avatarImage)
	})
	test('should render the avatar name', () => {
		render(<CardPost {...post} />)
		const p = screen.getByText(post.avatarName)
		expect(p).toBeTruthy()
	})
	test('should render the likes', () => {
		render(<CardPost {...post} />)
		const p = screen.getByText(post.likes)
		expect(p).toBeTruthy()
	})
})
