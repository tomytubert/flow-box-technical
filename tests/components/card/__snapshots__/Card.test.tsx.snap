// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Card /> should match the snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-eydnr8-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
    >
      <div
        class="MuiCardHeader-avatar css-1ssile9-MuiCardHeader-avatar"
      >
        <div
          aria-label="recipe"
          class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
        >
          <img
            class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="https://example.com/avatar.jpg"
          />
        </div>
      </div>
      <div
        class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
      >
        <span
          class="MuiTypography-root MuiTypography-body2 MuiCardHeader-title css-et1ao3-MuiTypography-root"
        >
          @example
        </span>
      </div>
    </div>
    <img
      alt="example"
      class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
      height="194"
      loading="lazy"
      src="https://example.com/image.jpg"
    />
    <div
      class="MuiCardContent-root css-20rqxf-MuiCardContent-root"
    >
      <div
        class="MuiBox-root css-syu8eu"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
          data-testid="FavoriteIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m12 21.35-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54z"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
        >
          100
        </p>
      </div>
      <p
        class="MuiTypography-root MuiTypography-body2 capitalize-first-letter css-r40f8v-MuiTypography-root"
      >
        example
      </p>
    </div>
  </div>
</div>
`;
